src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -313,6 +318,10 @@ protected function prepare_item_for_database( $request ) {
 			$config = array();
 			if ( isset( $request['styles'] ) ) {
```

Comment:
In this line of code,

```php
$validate_custom_css = $this->validate_custom_css( $request['styles']['css'] );
```

the assumption is both keys exist in the multidimensional array. However the `if` conditional is only checking for `'styles'` key and not `'css'` key.

I'd suggest to safe-guard this by checking for both:

```php
if ( isset( $request['styles']['css] ) ) {
```

If that's problematic, meaning it needs to fall into this branch, then a guard can be added to avoid an error from a missing key.

Reply:
Yeh, it needs to fall into that branch still if `css` is not set, so have added the extra guard.

Reply:
What should happen if `'css'` key is not set? Should it bail out? Should it error? Should it skip over that code and continue?

Reply:
I have changed the ordering of this to make it clearer, as there was no point in assigning `styles` to `$config` until after the validation of `css`. Does that make sense now?

Reply:
LGTM 👍 

marking as resolved.

Thanks @glendaviesnz!